pkg/cmd/infra/router/template.go

```diff
@@ -193,11 +201,66 @@ func (o *TemplateRouterOptions) Validate() error {
 
 // Run launches a template router using the provided options. It never exits.
 func (o *TemplateRouterOptions) Run() error {
+	var reloadCallbacks []func()
+	switch {
+	case o.MetricsType == "haproxy" && len(o.ListenAddr) > 0:
+		var timeout time.Duration
+		if t := util.Env("ROUTER_METRICS_HAPROXY_TIMEOUT", ""); len(t) > 0 {
+			d, err := time.ParseDuration(t)
+			if err != nil {
+				return fmt.Errorf("ROUTER_METRICS_HAPROXY_TIMEOUT is not a valid duration: %v", err)
+			}
+			timeout = d
+		}
+		var baseScrapeInterval time.Duration
+		if t := util.Env("ROUTER_METRICS_HAPROXY_BASE_SCRAPE_INTERVAL", ""); len(t) > 0 {
+			d, err := time.ParseDuration(t)
+			if err != nil {
+				return fmt.Errorf("ROUTER_METRICS_HAPROXY_BASE_SCRAPE_INTERVAL is not a valid duration: %v", err)
+			}
+			baseScrapeInterval = d
+		}
+		var serverThreshold int
+		if t := util.Env("ROUTER_METRICS_HAPROXY_SERVER_THRESHOLD", ""); len(t) > 0 {
+			i, err := strconv.Atoi(t)
+			if err != nil {
+				return fmt.Errorf("ROUTER_METRICS_HAPROXY_SERVER_THRESHOLD is not a valid integer: %v", err)
+			}
+			serverThreshold = i
+		}
+		var exported []int
+		if t := util.Env("ROUTER_METRICS_HAPROXY_EXPORTED", ""); len(t) > 0 {
```

Comment:
This is mostly the escape hatch in production, not yet intended for regular use.